### PR TITLE
use own terminal customized cursor shape

### PR DESCRIPTION
**Pull request by foruniverse**
first, this theme is absolutely great , but noticed that it changed my cursor shape in the terminal  while I preferred my old one, so I just add a "terminal default" option to implement that idea, ~~actually just a `if` statement.~~😁

### CODE REVIEW

File: typewritten.zsh [hunk 1]
```diff
@@ -150,7 +150,9 @@ tw_setup() {
 
   zmodload zsh/zle
   autoload -Uz add-zsh-hook
-  add-zsh-hook precmd tw_fix_cursor
+  if [ "$TYPEWRITTEN_CURSOR" != "terminal default" ]; then
```
Comment (reobin) [inline typewritten.zsh:153]: I think just `"terminal"` would be the more accurate term, semantically. If I change my cursor in my terminal emulator, then my terminal cursor is no longer `default`.

**Comment by reobin** [comment]
Hi @foruniverse !

Thank you for the PR 🎉 

Your need to let the terminal handle the cursor is totally valid. I'm happy you took the time to come up with this PR.

As a side note, the cursor fix handles a problem I am not sure the majority of the people have. It fixes a weird issue where exiting vim would set the terminal cursor to the one vim had, which is often a box. It's one of the first things I added to typewritten, so it might be time to actually think about it 😅.

One could argue that typewritten should not manage cursor by default. Something to discuss at another time

File: typewritten.zsh (hunk 1 above)
Comment (foruniverse) [reply typewritten.zsh:153]: of course, no problem, `terminal` is more accurate

**Comment by foruniverse** [comment]
> Hi @foruniverse !
> 
> Thank you for the PR 🎉
> 
> Your need to let the terminal handle the cursor is totally valid. I'm happy you took the time to come up with this PR.
> 
> Little backstory, the cursor fix handles a problem I am not sure the majority of the people have. It fixes a weird issue where exiting vim would set the terminal cursor to the one vim had, which is often a box.
> 
> It's one of the first things I added to typewritten, so it might be time to review it 😅.
> One could argue that typewritten should not manage cursor by default. Something to discuss at another time

that ' s ok 🤣,

File: typewritten.zsh (hunk 1 above)
Comment (reobin) [reply typewritten.zsh:153]: Cool, update it and we can merge. The documentation also needs to be updated for the `TYPEWRITTEN_CURSOR` section. You can update it, or leave it to me and I'll update your PR np.

**Review by reobin — APPROVED** [review]

**Comment by reobin** [comment]
@all-contributors please add @foruniverse for code

**Pull request merged**